### PR TITLE
#75 - Repository interface extracted

### LINKED ISSUE #75: Extract Repository interface
To support proxy PHP composer repository in future HTTP layer should depend on interface. Right now it depends on `Repository` class which is implementation of PHP composer repository storing data in ASTO storage.

### PR DESCRIPTION
**Pull request by olegmoz**
Closes #75 
Repository interface extracted

### CODE REVIEW

File: src/main/java/com/artipie/composer/Repository.java [hunk 1]
```diff
@@ -25,114 +25,37 @@
...
-import java.util.UUID;
 import java.util.concurrent.CompletableFuture;
 import java.util.concurrent.CompletionStage;
 
 /**
- * Class representing PHP Composer repository.
+ * PHP Composer repository.
  *
  * @since 0.1
  */
```
Comment (genryxy) [inline src/main/java/com/artipie/composer/Repository.java:36]: @olegmoz  also here `@since` tag should be changed

**Review by genryxy — CHANGES_REQUESTED** [review]
@olegmoz thanks, please see one comment

**Comment by olegmoz** [comment]
@genryxy thanks, updated

**Review by genryxy — APPROVED** [review]
@olegmoz thanks, approve

**Comment by 0crat** [comment]
Job `gh:artipie/composer-adapter#76` is not assigned, can't get performer

**Pull request merged**